oss-internship-2020/openjpeg/examples/convert_helper.cc

```diff
@@ -0,0 +1,311 @@
+
+// copies of a few library tools
+
+#include "convert_helper.h"
+
+const char* opj_version(void)
+{
+    return "2.3.1";
+}
+
+static int are_comps_similar(opj_image_t * image)
+{
+    unsigned int i;
```

Comment:
Just a nit, in newer C/C++ standards you can limit the visibility of the counter used in for(;;) to the scope of for(). It'll limit pollution of the variables to upper scopes.

E.g. for (int i = 1; ..

Reply:
Also, the type of numcomps seems to be OPJ_UINT32 which then typedefs to sth. else. If you include openjp2 header files I suggest using this type instead of int.